src/Qir/Tests/QIR-dynamic/qsharp/qir-test-assert.qs

```diff
@@ -1,15 +1,363 @@
+        AssertMeasurementProbability([PauliY, PauliY], [left, right],  One, 0.5, "A: Call failed", 1E-05);
+
+        H(right);       // Back to |1>
+        X(right);       // Back to |0>
+        H(right);       // |+>
+        // |left right> = |-+>
+        AssertMeasurement(           [PauliX, PauliX], [left, right],  One,      "G: Call failed");
+        // 50% probability in other Pauli bases:
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right], Zero, 0.5, "H: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliZ, PauliZ], [left, right],  One, 0.5, "I: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliY, PauliY], [left, right], Zero, 0.5, "J: Call failed", 1E-05);
+        AssertMeasurementProbability([PauliY, PauliY], [left, right],  One, 0.5, "K: Call failed", 1E-05);
     }
+
+    // Task 3.  |0000>+|1111>  or  |0011>+|1100> ?
+    @EntryPoint()
+    operation AssertMeas4qubits() : Unit {
+        use qubitIds = Qubit[4];        // |0000>
+
+        H(qubitIds[0]);                 // |+>, equally probable |0> and |1> in PauliZ basis.
+
+        if M(qubitIds[0]) == One {      // |+> collapses to |1> (-1 eigenvalue is observed)
+            X(qubitIds[1]);             //      |1100>
+        } else {
+            X(qubitIds[2]);             //      |0010>
+            X(qubitIds[3]);             //      |0011>
+        }
+
+        // Equally probable |1100> or |0011>.
+
+        AssertMeasurement(           [PauliZ, PauliZ], [qubitIds[0], qubitIds[1]], Zero,      "0: Call failed");  // |00> or |11>
+        AssertMeasurement(           [PauliZ, PauliZ], [qubitIds[1], qubitIds[2]],  One,      "1: Call failed");  // |01> or |10>
+        AssertMeasurement(           [PauliZ, PauliZ], [qubitIds[2], qubitIds[3]], Zero,      "2: Call failed");  // |00> or |11>
+        AssertMeasurementProbability([PauliZ, PauliZ], [qubitIds[0], qubitIds[1]], Zero, 1.0, "3: Call failed", 1E-05);   // |00> or |11>
+        AssertMeasurementProbability([PauliZ, PauliZ], [qubitIds[1], qubitIds[2]],  One, 1.0, "4: Call failed", 1E-05);   // |01> or |10>
+        AssertMeasurementProbability([PauliZ, PauliZ], [qubitIds[2], qubitIds[3]], Zero, 1.0, "5: Call failed", 1E-05);   // |00> or |11>
+    }
+
+    // Bell states (equal superposition of |00> and |11>, `(|10> + |01>) / SQRT(2)`):
+    // https://github.com/microsoft/Quantum/blob/0ec53c6efe09c0f725aae01648cd92377e2fcc99/samples/getting-started/measurement/Measurement.qs#L89
+    // "Quantum Computing : A Gentle Introduction", Example 3.2.1.  http://mmrc.amss.cas.cn/tlb/201702/W020170224608150244118.pdf
+
+    // States based on Bell state:
+    operation PrepareBellPair((applyX : Bool, applyZ : Bool), (left : Qubit, right : Qubit))  : Unit is Adj + Ctl {
+                            // Expects: |00>
+        H(left);            // Equally probable |0>  or |1>.  `(|0>  + |1> ) / SQRT(2)`
```

Comment:
The comments in this operation implicitly preference 𝑍-basis measurement. In particular, (|0⟩ + |1⟩) / √2 is not a mixture of |0⟩ and |1⟩, but a superposition such that if one measures in the 𝑋-basis instead, the result `Zero` is returned _deterministically_.

Put differently, measurement probabilities are only defined once you know what the measurement basis is. Since this operation only prepares a register into a particular state, it's inaccurate to refer to the probabilities of measurement outcomes without at least specifying the basis that those comments refer to.

Reply:
Thank you for correcting.
Changing the line 265 from `// Expects: |00>` to `// Expects: |00>. Comments below are in Z-basis.`